### PR TITLE
[Cleanup]: Fix Android Studio Warnings: 14 files changes

### PR DESCRIPTION
**Pull request by lakshay6907**
## Purpose / Description
_Describe the problem or feature and motivation_

## Fixes
* Fixes #

## Approach
_How does this change address the problem?_

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration (SDK version(s), emulator or physical, etc)

## Learning (optional, can help others)
_Describe the research stage_

_Links to blog posts, patterns, libraries or addons used to solve this problem_

## Checklist
_Please, go through these checks before submitting the PR._

- [ ] You have a descriptive commit message with a short title (first line, max 50 chars).
- [ ] You have commented your code, particularly in hard-to-understand areas
- [ ] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/AnkiActivity.kt [hunk 1]
```diff
@@ -1,5 +1,5 @@
 //noinspection MissingCopyrightHeader #8659
-@file:Suppress("LeakingThis") // fine - used as WeakReference
+@file:Suppress("LeakingThis", "SameReturnValue") // fine - used as WeakReference
```
Comment (mikehardy) [inline AnkiDroid/src/main/java/com/ichi2/anki/AnkiActivity.kt:2]: suppressing something at the file level shouldn't be done lightly, why is this necessary?
Note that the previous suppression had a comment which now is ambiguous as it doesn't relate to your new suppression, and the new suppression has no explanation

**Review by mikehardy — CHANGES_REQUESTED** [review]
Hi there!

I appreciate the contribution but have a question on one suppression, and it's important to note that not filling out the PR template is not very collaborative - seeing those come through with all defaults is a surprise for me on review

**Review by david-allison — CHANGES_REQUESTED** [review]
Looks good! A few reverts are required (probably a hint that too many warnings were taken on in this PR)
